src/hotspot/share/gc/serial/serialHeap.cpp

```diff
@@ -329,9 +329,9 @@ HeapWord* SerialHeap::mem_allocate_work(size_t size,
     {
       MutexLocker ml(Heap_lock);
       log_trace(gc, alloc)("SerialHeap::mem_allocate_work: attempting locked slow path allocation");
-      // Note that only large objects get a shot at being
-      // allocated in later generations.
-      bool first_only = !should_try_older_generation_allocation(size);
+
+      bool first_only = !should_try_older_generation_allocation(size)
+                     && is_long_enough_from_prev_gc_pause_end();
```

Comment:
Is the `first_only` actually `young_only`? It means that the allocation is only attempted in young generation?

Reply:
That's true. I believe Serial used to support >2 generations, so some generic names are left as is. Fixing it requires updating in multiple places, such as the method name `older_generation`, arg in `attempt_allocation`, etc. How about we do that invasive but pure refactoring in its own PR?

Reply:
Yes, such refactoring should be finished in other issues.